inc/Engine/Media/LazyloadSubscriber.php

```diff
@@ -309,11 +309,23 @@ public function insert_youtube_thumbnail_style() {
 	 * @return bool
 	 */
 	private function should_lazyload() {
-		if ( is_admin() || is_feed() || is_preview() || ( defined( 'REST_REQUEST' ) && REST_REQUEST ) || ( defined( 'DONOTLAZYLOAD' ) && DONOTLAZYLOAD ) ) {
-			return false;
-		}
-
-		if ( defined( 'DONOTROCKETOPTIMIZE' ) && DONOTROCKETOPTIMIZE ) {
+		// This filter is documented in inc/front/process.php.
+		$rocket_cache_search = apply_filters( 'rocket_cache_search', false );
+		if (
+			is_admin()
+			||
+			is_feed()
+			||
+			is_preview()
+			||
+			( is_search() && ! $rocket_cache_search )
+			||
+			rocket_get_constant( 'REST_REQUEST', false )
+			||
+			rocket_get_constant( 'DONOTLAZYLOAD', false )
+			||
+			rocket_get_constant( 'DONOTROCKETOPTIMIZE', false )
+		) {
```

Comment:
For cases where there is a lot of conditions to check, do we prefer splitting them into logical groups, or having them all in one place like this?

Reply:
It depends. Here there are a lot of them. What I often like to do is move them into a private method with a method name that _tells_ us what it's doing. Then the logic of the guard clause is encapsulated in its own method. What do you @Tabrisrp @engahmeds3ed think?

Reply:
You are right but this method do only two things:-
1. check these checks
2. check for the following fields if they are exist on $_GET `'fl_builder', 'et_fb', 'ct_builder'`
so it's simple function that almost do this check only.

Reply:
Also the `apply_filters()` isn't necessary if `is_admin()` or `is_feed()` or `is_preview()` are true or if `is_search()` is `false`.

We can refactor this list of guard clauses to be more performant and readable.

```suggestion
		if ( 
			rocket_get_constant( 'REST_REQUEST', false )
			||
			rocket_get_constant( 'DONOTLAZYLOAD', false )
			||
			rocket_get_constant( 'DONOTROCKETOPTIMIZE', false )
		{
			return false;
		}

		if ( is_admin() || is_feed() || is_preview() {
			return false;
		}

		// This filter is documented in inc/classes/Buffer/class-tests.php.
		if ( is_search() && ! (bool) apply_filters( 'rocket_cache_search', false ) ) {
			return false;
		}
```

Let's break it down:

1. Check the constants first. Why? If one of those constants is `true`, we bail out quicker. Result: more performant, i.e. less memory and faster execution.
2. Only check the filter once all the other guard clauses pass. Why? More performant. Filters are slower and use more memory.

Reply:
@Tabrisrp @engahmeds3ed What do you think of this approach 👆 ?

Reply:
Totally agree, this sequence order makes sense for me.

Reply:
@Tabrisrp Any concerns?

Reply:
I agree with this too

Reply:
@engahmeds3ed Can you take care of this refactor for us?

Reply:
@hellofromtonya Sure